src/poolman.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <algorithm>
+#include <vector>
+#include "poolman.h"
+#include "main.h"
+#include "util.h"
+#include "amount.h"
+#include "utiltime.h"
+
+using namespace std;
+
+static int64_t PoolmanLowWater, PoolmanHighWater;
+static bool janitorInit = false;
+static bool floatingFee = false;
+static CFeeRate origTxFee;
```

Comment:
A new class could be created to avoid new globals here. Or as said, this could be incorporated to the fee estimator.